### PR TITLE
11867 - fss tooling 2 digit year support in configuration

### PR DESCRIPTION
**Pull request by bhalcha13813**
11867 - FSS Tooling - Support 2 digit year option in configuration
12262 - FSS Tooling - Support Year for WeekNumber macro in configuration

### CODE REVIEW

File: file-share-service-desktop-client/Modules/Admin/JobViewModels/NewBatchJobViewModel.cs [hunk 1]
```diff
@@ -74,10 +74,27 @@ private string ExpandMacros(string value)
                         return "" + currentDateTimeProvider.CurrentDateTime.AddDays(dayOffset).Year;
                     }
                 },
+                {@"\$\(\s*now\.Year2\s*\)", (_) => "" + currentDateTimeProvider.CurrentDateTime.Year.ToString().Substring(2,2)},
```
[the remark this answers is not in the document]
Comment (bhalcha13813) [reply file-share-service-desktop-client/Modules/Admin/JobViewModels/NewBatchJobViewModel.cs]: modified as suggested

File: file-share-service-desktop-client/Modules/Admin/JobViewModels/NewBatchJobViewModel.cs [hunk 2]
```diff
@@ -74,10 +74,27 @@ private string ExpandMacros(string value)
...
+                        return "" + currentDateTimeProvider.CurrentDateTime.AddDays(dayOffset).Year.ToString().Substring(2,2);
+                    }
+                },
                 {
                     @"\$\(\s*now\.WeekNumber\s*\)",
                     (_) => "" + WeekNumber.GetUKHOWeekFromDateTime(currentDateTimeProvider.CurrentDateTime).Week
                 },
+                {
+                    @"\$\(\s*now\.WeekNumber\.Year\s*\)",
+                    (_) => "" + WeekNumber.GetUKHOWeekFromDateTime(currentDateTimeProvider.CurrentDateTime).Year
```
[the remark this answers is not in the document]
Comment (bhalcha13813) [reply file-share-service-desktop-client/Modules/Admin/JobViewModels/NewBatchJobViewModel.cs]: modified as suggested

File: file-share-service-desktop-client/Modules/Admin/JobViewModels/NewBatchJobViewModel.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (bhalcha13813) [reply file-share-service-desktop-client/Modules/Admin/JobViewModels/NewBatchJobViewModel.cs]: Added unit tests for new scenarios

**Review by crdandoukho — APPROVED** [review]
Thanks for changes. Looks good to me

**Review by crdandoukho — COMMENTED** [review]
I've asked Martin Rock-Evans if `now` and `now.AddDays' should be tolerant of surrounding whitespace like the other patterns are.

**Review by martyn-fewtrell — APPROVED** [review]
Looks good

**Pull request merged**